### PR TITLE
Cheil-Autotech's jsMotor01 has been added

### PR DESCRIPTION
**Pull request by Dalgona13**
Cheil-Autotech's jsMotor01 model has been added, which required the following changes:
1. Modified fingerprints.yml and init.lua files
2. Added .\cheil-auto\init.lua and .\test\test_zigbee_window_treatment_cheil.lua.

### CODE REVIEW

**Comment by lelandblue** [comment]
Hello @Dalgona13 ! Thank you for your interest. Please go to [developers.smartthings.com](http://developers.smartthings.com/) to continue the WWST certification process. If you have questions please reach out to [partners@smartthings.com](mailto:partners@smartthings.com) for more information.

Fingerprint changes to this repository will only be considered with a WWST Certification request.

File: drivers/SmartThings/zigbee-window-treatment/src/cheil-auto/init.lua [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+local function device_info_changed(driver, device, event, args)
+  if args.old_st_store.preferences.reverse ~= device.preferences.reverse then
+    set_shade_level_handler(driver, device, {args = { shadeLevel = 50 }})
+    device.thread:call_with_delay(2, function()
+      send_cmd_to_device(device, DP_ID_RESET_DIRECTION, DP_TYPE_ENUM, device.preferences.reverse and DP_VAL_REVERSE or DP_VAL_DIRECT)
+    end)
+  end
+end
+
+---------------------- Driver Template ------------------------h
```
Comment (wkhenon) [inline drivers/SmartThings/zigbee-window-treatment/src/cheil-auto/init.lua:199]:
```suggestion
---------------------- Driver Template ------------------------
```

File: drivers/SmartThings/zigbee-window-treatment/src/cheil-auto/init.lua [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+
+local capabilities = require "st.capabilities"
+local zb_zcl = require "st.zigbee.zcl"
+local zb_messages = require "st.zigbee.messages"
+local zb_const = require "st.zigbee.constants"
+local zb_data_types = require "st.zigbee.data_types"
+local zb_generic_body = require "st.zigbee.generic_body"
+local zb_window_preset_defaults = require "st.zigbee.defaults.windowShadePreset_defaults"
+
+--local Log = require "log"
```
Comment (wkhenon) [inline drivers/SmartThings/zigbee-window-treatment/src/cheil-auto/init.lua:18]: Is this comment still needed?

File: drivers/SmartThings/zigbee-window-treatment/src/cheil-auto/init.lua
Comment (greens) [inline drivers/SmartThings/zigbee-window-treatment/src/cheil-auto/init.lua:1]: Most of this code looks very similar, if not identical to the code in the `hanssem` driver. I'm wondering if you could simply modify the `can_handle` function of that driver to also apply to this device?

What are the differences between that handler and this one?

Comment (lelandblue) [reply drivers/SmartThings/zigbee-window-treatment/src/cheil-auto/init.lua:1]: Hello @Dalgona13 please review our feedback. We need to make sure the feedback is addressed and unit tests are passing before we can approve.

Thank you kindly.

**Review by lelandblue — COMMENTED** [review]
Hello @Dalgona13 If you are levering got from another public repo, such like what was done : [here](https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/pull/718/commits/093c0c6a585d1ace4ade8cb62900932cc69061bb)

Please properly credit that code to that author.

File: drivers/SmartThings/zigbee-window-treatment/src/cheil-auto/init.lua (hunk 2 above)
Comment (Dalgona13) [reply drivers/SmartThings/zigbee-window-treatment/src/cheil-auto/init.lua:18]: Deleted it.

File: drivers/SmartThings/zigbee-window-treatment/src/cheil-auto/init.lua (hunk 1 above)
Comment (Dalgona13) [reply drivers/SmartThings/zigbee-window-treatment/src/cheil-auto/init.lua:199]: Deleted it/

File: drivers/SmartThings/zigbee-window-treatment/src/cheil-auto/init.lua
Comment (Dalgona13) [reply drivers/SmartThings/zigbee-window-treatment/src/cheil-auto/init.lua:1]: There are difference in command number compared to the Hanssem driver. These are due to the firmware of motor controller has been updated.

Comment (greens) [reply drivers/SmartThings/zigbee-window-treatment/src/cheil-auto/init.lua:1]: My recommendation would be to refactor the hanssem driver to handle both devices. The different constants could be handled by putting them behind a function that checks the mfr/model.

```
local constants = function(model)
  if model == "jsMotor01" then
    return {...}
  elseif mode == "TS0601" then
    return {...}
  end
end
```

Something like that.

There's too much duplicated code here. It doesn't make sense to repeat it in two separate files.

Comment (greens) [reply drivers/SmartThings/zigbee-window-treatment/src/cheil-auto/init.lua:1]: I did like your usage of the `ReceiveHandlerSet` map and I think that's a more elegant solution as well.

Comment (Dalgona13) [reply drivers/SmartThings/zigbee-window-treatment/src/cheil-auto/init.lua:1]: Thanks for your guidance, I've understood the policy and will be follow it. Should I cancel this pull request and create a new one?

**Comment by lelandblue** [comment]
@Dalgona13 There are still open items on this PR that need to be resolved, including https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/pull/1010#pullrequestreview-1723145430

File: drivers/SmartThings/zigbee-window-treatment/src/cheil-auto/init.lua
Comment (lelandblue) [reply drivers/SmartThings/zigbee-window-treatment/src/cheil-auto/init.lua:1]: Please just continue to edit this one

**Pull request closed without merging**